Java/search/Fibonacci_search.java

```diff
@@ -0,0 +1,85 @@
+// Find a Fibonacci number that is greater than or equal to the size of the array in which we are searching for the key.
+// Compare the key with the predecessor of the Fibonacci number obtained in Step 1 and store it in index.
+// If the key and array element at index are equal then the key is present at position index + 1.        
+// If the key is less than array element at index, then search the left sub-tree to index.
+// If the given key is greater than the array element at index, then search the right sub-tree to index.  
+// If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size. After each iteration the size of array array_size is reduced.
```

Comment:
Rest looks good to merge.
Your last point in the comment exceed the range which looks not good. Only change needed, Split your last comment into two parts:
// If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size.
// After each iteration the size of array array size is reduced.

Reply:
ok